cpp/doc/DOCUMENTATION.md

```diff
@@ -0,0 +1,563 @@
+[doxygen_groups.h](../include/doxygen_groups.h) header file. The `doxygen_groups.h` file does not
+need to be included in any other source file, because the definitions in this file are used only by
+the doxygen tool to generate groups in the _Modules_ page. Modify this file only to add or update
+groups. The existing groups have been carefully structured and named, so new groups should be added
+thoughtfully.
+
+When creating a new API, specify its group using the
+[@ingroup](http://www.doxygen.nl/manual/commands.html#cmdingroup) tag and the group reference id
+from the [doxygen_groups.h](../include/doxygen_groups.h) file.
+
+```c++
+namespace cuspatial {
+
+/**
+ * @brief ...
+ *
+ * @ingroup distance
+ *
+ * @tparam...
+ * @param ...
+ * @return ...
+ */
+template <class Cart2dItA, class Cart2dItB, class OutputIt>
+OutputIt pairwise_point_distance(Cart2dItA points1_first, ...);
+
+}  // namespace cuspatial
+```
+
+You can also use the `@addtogroup` with a `@{ ... @}` pair to automatically include doxygen comment
+blocks as part of a group.
+
+```c++
+namespace cuspatial {
+/**
+ * @addtogroup coordinate_transform
+ * @{
+ */
+
+/**
+ * @brief ...
+ *
+ * @param ...
+ * @return ...
+ */
+template <class Cart2dItA, class Cart2dItB, class OutputIt>
+OutputIt pairwise_point_distance(Cart2dItA points1_first, ...);
+
+/** @} */
+}  // namespace cuspatial
+```
+
+This just saves adding `@ingroup` to individual doxygen comment blocks within a file. Make sure a
+blank line is included after the `@addtogroup` command block so doxygen knows it does not apply to
+whatever follows in the source code. Note that doxygen will not assign groups to items if the
+`@addtogroup` with `@{ ... @}` pair includes a namespace declaration. So include the `@addtogroup`
+and `@{ ... @}` between the namespace declaration braces as shown in the example above.
+
+Summary of groups tags
+| Tag/Command | Where to use |
+| ----------- | ------------ |
+| `@defgroup` | For use only in [doxygen_groups.h](../include/doxygen_groups.h) and should include the group's title. |
+| `@ingroup` | Use inside individual doxygen block comments for declaration statements in a header file. |
+| `@addtogroup` | Use instead of `@ingroup` for multiple declarations in the same file within a namespace declaration. Do not specify a group title. |
+| `@{ ... @}` |  Use only with `@addtogroup`. |
+
+## Build Doxygen Output
+
+We recommend installing Doxygen using conda (`conda install doxygen`) or a Linux package manager
+(`sudo apt install doxygen`). Alternatively you can
+[build and install doxygen from source](http://www.doxygen.nl/manual/install.html).
+
+To build the libcuspatial HTML documentation simply run the `doxygen` command from the `cpp/doxygen`
+directory containing the `Doxyfile`. The libcuspatial documentation can also be built using
+`make docs_cuspatial` from the cmake build directory (e.g. `cpp/build`). Doxygen reads and processes
```

Comment:
I actually have issue executing this command. I think it's also what leads to `docs-cuspatial-cpp` not working in my compose. @trxcllnt

Reply:
```
make docs_cuspaital
make: *** No rule to make target 'docs_cuspaital'.  Stop.
```

Reply:
You have a typo. Did you try `make docs_cuspatial`?

Reply:
Removing this. Doesn't seem to exist.

Reply:
Re-added and clarified, and added the missing targets to the CMake. Now need a cmake approval.